packages/frontend/src/utils/wallet.js

```diff
@@ -149,6 +151,13 @@ class Wallet {
         this.accountId = localStorage.getItem(KEY_ACTIVE_ACCOUNT_ID) || '';
     }
 
+    static KEY_TYPES = {
```

Comment:
Could this just be a constant independent of the `Wallet` class? The less this single class is responsible for, the better 😄

Reply:
hmm I was intentional with this one because I felt like the wallet _should_ have the concept of defined `KEY_TYPES` and I wanted to encourage this pattern. Agreed that this class is chunky though maybe some other things can be moved out

Reply:
Sure it's a concept related to the wallet, but you could make that argument about most constants defined in the codebase. Since these constants are useful outside of the wallet class (i.e. not strictly internal to the class), I really think it makes more sense to define them independent of the class.

The thinking that anything related to the wallet needs to be part of this single class is part of the reason that this module is 1,000 lines long 😄 . The more the class is extended with new methods and properties, the harder it becomes to break it up into sane modules with well-defined responsibilities. By decoupling these definitions from the class instance or the class itself, we're using the implicit convention we already employ for constants which is to define them as plain objects in module scope.

Reply:
It's not that it's related to the wallet, it's that this is a general enough concept to require a centralized store. We don't want to encourage other definitions of this value and we want to have it be in scope of the wallet's account and key management logic to help be explicit about the types of keys we handle. It is used outside of this class but it's arguable that it shouldn't be and we should use pre defined methods and only access it and do comparisons that way.  It is probably ideal that this class should be split into multiple classes implementing e.g. the signer override for ledger but I would consider that outside of the scope of this ticket. For now I'd like to keep it here until that refactor.

We also shouldn't use composition here imo. This concept is primitive to wallet keys and our key store and ideally should be a static member not an instance variable. Looking through this class, I would probably move out the logic calling the contract helper out first and then other logic like linkdrop specific logic but we do need a class to manage the wallet key store, our connection with `near-api-js` and other account related local storage logic (which ideally shouldn't exist) and this variable is right at home there.

Reply:
It's not that it's related to the wallet, it's that this is a general enough concept to require a centralized store. We don't want to encourage other definitions of this value and we want to have it be in scope of the wallet's account and key management logic to help be explicit about the types of keys we handle. It is used outside of this class but it's arguable that it shouldn't be and we should use pre defined methods and only access it and do comparisons that way. It is probably ideal that this class should be split into multiple classes implementing e.g. the signer override for ledger but I would consider that outside of the scope of this ticket. For now I'd like to keep it here until that refactor.

We could possibly use composition here but this is over engineering since other references should still be to the centralized `Wallet.KEY_TYPES` and not to this class: 
```js
class WalletKeyTypes {
  static KEY_TYPES;
  static isFullAccessKeyInfoView;
  static isLedgerKeyInfoView;
  static isMultisigKeyInfoView;
}

class Wallet {
 static KEY_TYPES = WalletKeyTypes;
}
```

This concept is primitive to wallet keys and our key store and ideally should be a static member not an instance variable. Looking through this class, I would probably move out the logic calling the contract helper out first and then other unrelated logic e.g. linkdrop specific logic but we do need a class to manage the wallet key store, our connection with `near-api-js` and other account related local storage logic (which ideally shouldn't exist) and this variable is right at home there. There is probably a great class separation for that too but it's not the above and we haven't reached a point where we need to make that separation to the key store management class. Maybe it would be something like 3 different `WalletKey` classes with a single keyType member. 

Do share a better design of you see one please 😄

Reply:
@esaminu all I'm suggesting is that the `KEY_TYPES` object doesn't need to be defined in the class. I don't see the value in needing to import the wallet singleton or class to access a these string constants. I really don't think it needs to be as complex as what you're outlining with class composition, just an object exported from a module with similar constants. This is a fairly conventional approach to reusing constants with shared meaning across the codebase, e.g.:

`constants.js`
```js
export const KEY_TYPES = {
    ...
};
```